### PR TITLE
fix(ext/console): fix inspecting iterators error.

### LINKED ISSUE #19776: deno repl: fails to inspect MapIterator or SetIterator
```console
> new Map()[Symbol.iterator]()
[Internal Formatting Error] TypeError: Cannot read properties of undefined (reading 'length')
    at formatSetIterInner (ext:deno_console/01_console.js:2215:53)
    at formatIterator (ext:deno_console/01_console.js:1504:10)
    at formatRaw (ext:deno_console/01_console.js:1038:14)
    at formatValue (ext:deno_console/01_console.js:721:10)
    at Object.inspectArgs (ext:deno_console/01_console.js:3114:28)
    at <anonymous>:3:55
```

same goes for `new Set()[Symbol.iterator]()`

### PR DESCRIPTION
**Pull request by ghosind**
Fixes #19776 and #20676 .

Node implements `previewEntries` as C++ binding, so I moved this function to internal binding.

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks for the PR @ghosind, it appears that this change caused an actual test failure in Node.js compat tests. Could you please take a look?

**Comment by ghosind** [comment]
Hi @bartlomieju ,
`Console.table` calls `console.log` and consumes the iterator. So, `console.table` test cases that print iterators will be failures. However, it'll throw an error and terminate the test case before this change. I've removed this log and no test error now.

**Comment by bartlomieju** [comment]
@crowlKats please review

File: ext/console/01_console.js [hunk 1]
```diff
@@ -1494,8 +1495,7 @@ function getIteratorBraces(type, tag) {
 const iteratorRegExp = new SafeRegExp(" Iterator] {$");
 function formatIterator(braces, ctx, value, recurseTimes) {
   // TODO(wafuwafu13): Implement
```
Comment (crowlKats) [inline ext/console/01_console.js:1497]:
```suggestion
```

File: ext/console/01_console.js [hunk 2]
```diff
@@ -1705,15 +1705,13 @@ function formatWeakCollection(ctx) {
 
 function formatWeakSet(ctx, value, recurseTimes) {
   // TODO(wafuwafu13): Implement
```
Comment (crowlKats) [inline ext/console/01_console.js:1707]:
```suggestion
```

File: ext/console/01_console.js [hunk 3]
```diff
@@ -1705,15 +1705,13 @@ function formatWeakCollection(ctx) {
...
-  // const entries = previewEntries(value);
-  const entries = value;
+  const entries = previewEntries(value);
   return formatSetIterInner(ctx, recurseTimes, entries, kWeak);
 }
 
 function formatWeakMap(ctx, value, recurseTimes) {
   // TODO(wafuwafu13): Implement
```
Comment (crowlKats) [inline ext/console/01_console.js:1713]:
```suggestion
```

**Review by crowlKats — APPROVED** [review]
Thanks, LGTM

**Comment by mmastrac** [comment]
This PR unfortunately broke benchmarks on main:

```
running 1 test
thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/deno_core-0.229.0/runtime/jsruntime.rs:788:42:
called `Result::unwrap()` on an `Err` value: Specifier "ext:deno_node/internal_binding/util.ts" was not passed as an extension module and was not included in the snapshot.
stack backtrace:
```

**Pull request merged**